Realm/RLMProperty.mm

```diff
@@ -292,18 +294,23 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
         }
     }
 
-    if (![self setTypeFromRawType]) {
+    void (^throwForPropertyName)(NSString *) = ^(NSString *propertyName){
         @throw RLMException(@"Can't persist property '%@' with incompatible type. "
-                            "Add to ignoredPropertyNames: method to ignore.", self.name);
+                            "Add to Object.ignoredProperties() class method to ignore.", propertyName);
+    };
+
+    if (![self setTypeFromRawType]) {
+        throwForPropertyName(self.name);
     }
 
-    // convert type for any swift property types (which are parsed as Any)
-    if (_type == RLMPropertyTypeAny) {
+    if (_type == RLMPropertyTypeAny && [obj valueForKey:_name] != nil) {
         if ([[obj valueForKey:_name] isKindOfClass:[NSString class]]) {
+            // convert type for any swift property types (which are parsed as Any)
             _type = RLMPropertyTypeString;
+        } else if (![[obj valueForKey:_name] isKindOfClass:[RLMListBase class]] && ![[obj valueForKey:_name] isKindOfClass:[RLMOptionalBase class]]) {
```

Comment:
It's weird to verify that the property is a list or optional, but leave it as Any, when a line above the type is being changed for Strings. Could use a comment as to why (I think it's because List/RealmOptional properties only get reported to obj-c with Swift 1.2 and not 2+ so we can't rely on that).